src/Features/CSharp/Portable/ConvertSwitchStatementToExpression/ConvertSwitchStatementToExpressionDiagnosticAnalyzer.AnalysisResult.cs

```diff
@@ -79,7 +80,8 @@ public static AnalysisResult Match(AnalysisResult left, AnalysisResult right)
                 if (left.IsAssignment && right.IsAssignment)
                 {
                     // Assignments only match if they have the same set of targets
-                    if (left._assignmentTargets.SequenceEqual(right._assignmentTargets))
+                    if (left._assignmentTargets.SequenceEqual(right._assignmentTargets,
+                        (leftNode, rightNode) => leftNode.IsEquivalentTo(rightNode)))
```

Comment:
Note: With symbols we might get false positive in case there's some arguments. Considering both seems like overkill. Another option would be symbols + comparing arguments syntactically, but I think it wouldn't worth it. So I just switched to expressions.

Reply:
👍